src/cargo/util/toml.rs

```diff
@@ -120,7 +120,34 @@ pub fn to_manifest(contents: &[u8],
         human(e.to_string())
     }));
 
-    let pair = try!(manifest.to_manifest(source_id, &layout, config));
+    let opt_stdlib_reference = if
+        try!(config.get_bool("build-stdlib")).map(|v| v.val).unwrap_or(false)
+    {
+        match config.rustc_info().version_get("commit-hash") {
+            Some("unknown") | None => None,
+            Some(hash) => Some(GitReference::Rev(hash.to_string())),
+        }.or_else(|| match config.rustc_info().version_get("release") {
+            Some("unknown") | None => None,
+            Some(release) => Some(GitReference::Tag(release.to_string())),
+        })
+    } else {
+        None
+    };
+
+    let opt_stdlib_repo = match opt_stdlib_reference {
+        None => None,
+        Some(reference) => {
+            let config = try!(config.get_string("stdlib-repo"));
+            let url = config.as_ref().map(|s| &s.val[..])
+                .unwrap_or("http://github.com/rust-lang/rust.git");
```

Comment:
What is the actual URL? (I hate how hard it is to browse https://static.rust-lang.org.) If the nightly repos are up by date, but `rustc -vV` only gives us the commit hash (both those statements might be false :)), this will be a lot harder.

I'm generally interested in fixing the perf problem with Cargo, but at this point I just want to see explicit stdlib deps in some form and am not too picky :).